src/PIL/GifImagePlugin.py

```diff
@@ -144,7 +144,7 @@ def _seek(self, frame):
             self.disposal_method = 0
         else:
             # ensure that the previous frame was loaded
-            if not self.im:
+            if self.tile:
```

Comment:
Okay, so I realise now that this commit is solving the same problem as #3716.

To compare the two solutions, #3716 adds a new load every time a backwards seek is made. This solution adds a new load for each frame seeked forward.

Adding some print statements to demonstrate, I get -

```python
>>> from PIL import Image
>>> im = Image.open("Tests/images/iss634.gif")
>>> im.seek(10)
This adds a new load, when seeking forward to 2
This adds a new load, when seeking forward to 3
This adds a new load, when seeking forward to 4
This adds a new load, when seeking forward to 5
This adds a new load, when seeking forward to 6
This adds a new load, when seeking forward to 7
This adds a new load, when seeking forward to 8
This adds a new load, when seeking forward to 9
This adds a new load, when seeking forward to 10
>>> im.seek(5)
#3716 adds a new load, when seeking back to 5
This adds a new load, when seeking forward to 1
This adds a new load, when seeking forward to 2
This adds a new load, when seeking forward to 3
This adds a new load, when seeking forward to 4
This adds a new load, when seeking forward to 5
```

The tests added in this PR still pass with the change from #3716. So I'm in favour of #3716. Feel free to add your thoughts.

Reply:
I don't entirely follow (don't have time to dig into it again right now), but if you seek forward in a GIF, you need to decode each frame, since the decoding is incremental.

Reply:
No worries if you don't have time right now, and If you think that this needs to be there instead of or regardless of my change, that's fine. I'm just saying that the tests don't show evidence of that.

Reply:
Ok, I've separated this out into #5386